[User Guide] Add New Topic For Locating Examples

hcnagaraj-e opened the pull request:
### What does this Pull Request accomplish?
Adding topic for locating STL examples

### Why should this Pull Request be merged?
Need GitHub page document for locating and navigating through STL examples

### What testing has been done?

- [x] Ensured linting done
- [x] Using VS code verified that document is rendering properly
- [x] Content has appropriate links
- [x] verified navigation for index.md

Mattjet27 commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+Code snippet examples demonstrate how to use specific API methods or implement coding concepts.
+
+Example use cases are represented by individual static methods within separate static class files that are organized by high-level categories, such as instrument abstraction and data abstraction.
+The example methods are not designed to run independently. Example methods have contextual requirements that are described within the example method summary or code comments.
```
Please make these bulleted to match the formatting of ni.com documentation. 

Also, please add the following note to the end of this section:
```md
> [!NOTE]
> Code snippet style examples were the only example style available for v24.5.0, v24.5.1, and v25.0.0 releases of the Semiconductor Test Library.
```

hcnagaraj-e replied:
Updated as suggested

hcnagaraj-e commented:
> @hcnagaraj-e, this is my first pass review of this. Note, we will need this PR to also update the _Examples and Source Code_ section of the index.md file. I expect that section to be rewritten as follows:
> 
> ```
> For examples, refer to [Locating Examples](UserGuide/LocatingExamples.md).
> 
> For source code, visit the [semi-test-library-dotnet](https://github.com/ni/semi-test-library-dotnet) repository in GitHub.
> ```

Updated index.md file and renamed new document to LocatingExamples.md

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+# Locating Semiconductor Test Library Examples
+
+Semiconductor Test Library examples are located here: [Semi-test-library-dotnet/Examples](https://github.com/ni/semi-test-library-dotnet/tree/main/Examples/source). 
```
Updated as suggested.
I have also change to STS software installer

_The version of the examples included in the STS Software install**er** will match their state at the time_

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+# Locating Semiconductor Test Library Examples
+
+Semiconductor Test Library examples are located here: [Semi-test-library-dotnet/Examples](https://github.com/ni/semi-test-library-dotnet/tree/main/Examples/source). 
+There are different styles of examples available, as described below.
```
Updated as suggested

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+It can be executed standalone, outside of the TestStand Sequence Editor, if the criteria defined by the provided context is met.
+Runs in offline mode with simulated hardware or with actual hardware depending on the criteria defined by the provided context.
+
+[Semi-test-library-dotnet/Examples/Standalone](https://github.com/ni/semi-test-library-dotnet/tree/main/Examples/source/Standalone)
```
Removed hyperlink from all sections

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+Each test program contains fully-documented code implementations. Each test method has code-level summary documentation.
+Test programs run in offline mode with simulated hardware.
+
+[Semi-test-library-dotnet/Examples/TestPrograms](https://github.com/ni/semi-test-library-dotnet/tree/main/Examples/source/TestPrograms)
```
Removed hyperlink from all sections

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+Any requirements for understanding or running the example are documented in an accompanying README.md. Context is also provided in sequence comments or inline code statements.
+Example sequences run in offline mode with simulated hardware or with actual hardware, depending on the documented requirements.
+
+[Semi-test-library-dotnet/Examples/Sequence](https://github.com/ni/semi-test-library-dotnet/tree/main/Examples/source/Sequence)
```
Removed hyperlink from all sections

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+
+Test Program examples demonstrate mock test programs for testing specific DUT types.
+
+The supported DUT type, requirements, and assumptions are documented in an accompanying README.md.
```
Removed hyperlink from all sections

hcnagaraj-e commented on docs/UserGuide/SemiconductorTestLibraryExamples.md:
```diff
@@ -0,0 +1,52 @@
+
+Standalone style examples demonstrate how to program instrumentation or implement code-level concepts.
+
+Any assumed context required to run or understand the example is documented within an accompanying README.md file and via comments in-line with the code statements and method summaries.
```
Removed hyperlink from all sections

Mattjet27 commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,71 @@
+> [!NOTE]
+> Test program style examples are available as of the v25.5 release of the Semiconductor Test Library (STL). Unless documented otherwise, they can be leveraged with previous STL versions. Refer to the documentation accompanying the specific example for details.
+
+
```
There is an extra line here above `## Standalone Style`. There should only be one line between this header and the note above it.

Mattjet27 commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,71 @@
+
+> [!NOTE]
+> There are no standalone style examples that have been released to date. This style of example is being considered for future releases.
+
```
There appears to be an extra line at the end of the file. There should only be one blank line at the end of the file after the note ends.

hcnagaraj-e replied:
Removed extra line.

Mattjet27 reviewed (changes requested):
@hcnagaraj-e, in addition to resolving the [comment](https://github.com/ni/semi-test-library-dotnet/pull/335#discussion_r2333627678) above from my last review, I have two new minor comments. Other than these, this is ready for tech writer review.

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+# Locating Examples
+
+Examples for how to use the Semiconductor Test Library (STL) are available on GitHub and are also included as part of the STS Software install.
```
"and are also installed with STS Software."

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+
+Reference the GitHub location for the latest version of the examples or to browse the example code source online.
+When on a system with STS Software installed, reference the examples from their installed location.
+The version of the examples included in the STS Software install will match their state at the time of the specific STL release used by the STS Software version.
```
Here's a rewrite with a bunch of small changes:

The installed version of an example matches the state of the example at the time of the STL release consumed by the installed STS Software version.

hcnagaraj-e replied:
@nstokes-nati Updated as suggested. 
@Mattjet27 Please provide your suggestion, if you any concern with newly suggested wording.

Mattjet27 replied:
@hcnagaraj-e, no concerns.

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+When on a system with STS Software installed, reference the examples from their installed location.
+The version of the examples included in the STS Software install will match their state at the time of the specific STL release used by the STS Software version.
+Refer to the [STS Software Version Compatibility](https://ni.github.io/semi-test-library-dotnet/UserGuide/Overview.html#sts-software-version-compatibility) table for more details.
+
```
We didn't include all four types in the ni.com documentation, just code, sequence, and test programs. https://ni-staging.zoominsoftware.io/docs/en-US/bundle/semiconductor-test-library/page/stl-examples.html

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+
+Standalone style examples demonstrate how to program instrumentation or implement code-level concepts.
+
+- Any assumed context required to run or understand the example is documented within an accompanying README.md file and via comments in-line with the code statements and method summaries.
```
Copy from Sequence Examples list item: - Any requirements for understanding or running the example are documented in an accompanying README.md. Context is also provided in method summaries or inline code statements.

hcnagaraj-e replied:
[nstokes-nati](https://github.com/nstokes-nati) Updated as suggested

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+> [!NOTE]
+> Test program style examples are available as of the v25.5 release of the Semiconductor Test Library (STL). Unless documented otherwise, they can be leveraged with previous STL versions. Refer to the documentation accompanying the specific example for details.
+
+## Standalone Style
```
Standalone Examples

Mattjet27 replied:
@hcnagaraj-e, if we are going change style to Example for the section headers, then you forgot to update the list in top section to do the same.

hcnagaraj-e replied:
I probably missed saving the file after editing the subheadings. Anyways, now that we are removed section and bullet point, no action needed. I am marking it as resolved.

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+Standalone style examples demonstrate how to program instrumentation or implement code-level concepts.
+
+- Any assumed context required to run or understand the example is documented within an accompanying README.md file and via comments in-line with the code statements and method summaries.
+- It can be executed standalone, outside of the TestStand Sequence Editor, if the criteria defined by the provided context is met.
```
If the documented requirements are met, standalone examples can be executed independently outside the TestStand Sequence Editor.

hcnagaraj-e replied:
@nstokes-nati Updated as suggested FYI: @Mattjet27

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+
+- Any assumed context required to run or understand the example is documented within an accompanying README.md file and via comments in-line with the code statements and method summaries.
+- It can be executed standalone, outside of the TestStand Sequence Editor, if the criteria defined by the provided context is met.
+Runs in offline mode with simulated hardware or with actual hardware depending on the criteria defined by the provided context.
```
Depending on the individual example requirements, a standalone example may require physical hardware or run in offline mode with simulated hardware.

hcnagaraj-e replied:
@nstokes-nati Updated as suggested

nstokes-nati commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+- It can be executed standalone, outside of the TestStand Sequence Editor, if the criteria defined by the provided context is met.
+Runs in offline mode with simulated hardware or with actual hardware depending on the criteria defined by the provided context.
+
+> [!NOTE]
```
I would suggest deleting the entire section until these examples are added. Otherwise: 

Standalone examples are not yet available but will be included in future releases.

hcnagaraj-e replied:
@nstokes-nati I think the idea of this GitHub document is to make it relevant at all the time (throughout the dev cycle). Not just linked to current release. Let's say, in next 1 or 2 months, if we add Standalone style example, document should still be relevant to current state. 

@Mattjet27 Please correct me if I am wrong.

Mattjet27 replied:
@hcnagaraj-e, please see my reply to the above [comment](https://github.com/ni/semi-test-library-dotnet/pull/335#discussion_r2337361231).

Mattjet27 commented on docs/UserGuide/LocatingExamples.md:
```diff
@@ -0,0 +1,69 @@
+
+## Standalone Style
+
+Standalone style examples demonstrate how to program instrumentation or implement code-level concepts.
```
@hcnagaraj-e and @nstokes-nati, perhaps this should be "Standalone Examples" to match the other sections.
